autolab/Autolab
# Implement Backwards Compatibility to Per-user Handin Subdirectories

## Description


This patch to #1831 provides backwards-compatibility to submissions created with the old submissions file structure (i.e. all submissions / autograde output / annotated pdfs are within the handin directory). This allows users to still access and download these associated files within Autolab, but any future submissions created will use the new handin structure.

## Motivation and Context



This patch has become necessary as the move to per-user handin subdirectories has become higher priority due to multiple instructors facing issues when the number of files within the handin directory exceeds the AFS limit due to thousands of student submissions. However, #1831 was created for 3.0 in mind, but due to the new shift in priority, this fix will also apply to 2.x versions. Therefore, backwards compatibility is necessary for current deployment as to not suddenly break the countless number of previous assessments using the old submission handin structure.

## How Has This Been Tested?




1. Checkout master
2. Submit to:
    - Autograded assessment
    - PDF assessment
    - tar / "archive" assessment
3. For PDF assessment, download annotated version
4. Verify submission structure:
    - Archive
![Screenshot 2023-03-27 at 10 49 48 AM](https://user-images.githubusercontent.com/25730111/227976961-d679d350-bab8-4d5a-955c-dbc92f5fcd38.png)
    - Autograded
![Screenshot 2023-03-27 at 10 49 58 AM](https://user-images.githubusercontent.com/25730111/227976963-84e936f5-2141-49ce-a7d6-47058c227843.png)
    - PDF
![Screenshot 2023-03-27 at 10 50 05 AM](https://user-images.githubusercontent.com/25730111/227976964-372b406a-c8c7-4740-be6b-0c9d21d367f5.png)
6. Switch to this branch
7. Submit to:
    - Autograded assessment (submit different tar/file to see difference easily)
    - PDF assessment (submit different PDF)
      - make sure to download annotated again
    - tar / "archive" assessment (submit different archive)
8. Confirm file locations of the two submissions for each (there should now be a subdirectory for the 2nd set of submissions)
    - Archive
![Screenshot 2023-03-27 at 10 54 10 AM](https://user-images.githubusercontent.com/25730111/227978135-516f66fb-8493-430a-8b9c-b38716c1e362.png)
    - Autograded
![Screenshot 2023-03-27 at 10 54 27 AM](https://user-images.githubusercontent.com/25730111/227978137-e199bb3b-6f74-47f4-b51f-12f91d5f55b4.png)
    - PDF
![Screenshot 2023-03-27 at 10 54 34 AM](https://user-images.githubusercontent.com/25730111/227978140-9cb02c69-7f93-464e-ac00-c251042c3631.png)
9. For autograded assessment:
    - Try to download both submissions, see that both download correctly
    - View both submissions and Autograded Output
    - Ensure download file works for both submissions
    - Verify regrade works for both and that files don't move
    - Delete old submission, see that it gets archived
    - Delete new submission, see that it gets archived
![Screenshot 2023-03-27 at 11 01 37 AM](https://user-images.githubusercontent.com/25730111/227980546-e5000778-821b-474c-be5c-4790ddafb05d.png)
10. For Archive:
    - Try to download both submissions, see that both download correctly
    - View both submissions
    - Ensure download file works for both submissions
    - Delete old submission, see that it gets archived
    - Delete new submission, see that it gets archived
  
![Screenshot 2023-03-27 at 11 06 44 AM](https://user-images.githubusercontent.com/25730111/227981838-c5e40719-f34b-42df-83dd-5c4a6da2a6d2.png)

 11. For PDF:
    - view both submissions
    - Ensure download, download annotated work for both submissions
    - Delete old submission, see that it gets archived
    - Delete new submission, see that it gets archived
  
![Screenshot 2023-03-27 at 11 06 20 AM](https://user-images.githubusercontent.com/25730111/227981902-b6390e08-dc07-4a63-b8fd-f6861b4ea80a.png)

## Types of changes


## Checklist:
